builder/alicloud/ecs/run_config_test.go

```diff
@@ -68,7 +69,7 @@ func TestRunConfigPrepare_UserData(t *testing.T) {
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
-	defer tf.Close()
```

Comment:
Are we able to remove a file that hasn't been closed, or do we need both commands?

Reply:
Oh dear! You're right... we need the `tf.Close()` as well. Looking [here](https://github.com/hashicorp/packer/blob/master/builder/vmware/iso/builder_test.go#L332-L333) would have shown me the correct way!

I don't know why, but I'd kind of assumed `os.Remove` would just 'magically' handle everything for me! I'll work through and fix...